implementations/prometheus/e2e_benchmark_test.go

```diff
@@ -0,0 +1,105 @@
+package prometheus
+
+import (
+	"context"
+	"github.com/grafana/walqueue/types"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/prometheus/prompb"
+	"github.com/prometheus/prometheus/storage"
+	"github.com/stretchr/testify/require"
+)
+
+func BenchmarkE2E(b *testing.B) {
+	/*
+		This adds 10_000 prometheus metrics and checks them. 70% of the allocs are building the test series. 14k are the actual work.
```

Comment:
Is this still accurate? I saw the recommendation in slack to pause the benchmark for setup allocations.

```
b.StopTimer()
		// allocate here
		b.StartTimer()
		```

Reply:
I have now fixed this properly! I should not have been using the parent functions testing var but instead using the specific runs testing var.